### PR TITLE
[ISSUE #1870] Add Support for Java 9+ and compatible to previous Java version

### PR DESCRIPTION
**Pull request by HashZhang**
## What is the purpose of the change

Add Support for Java 9+ and compatible to previous Java version

## Brief changelog

Add JVM Flag compatible for Java 9+ and configure JVM Flags for various Java version

### CODE REVIEW

**Comment by HashZhang** [comment]
Noted, will be done

**Comment by imaffe** [comment]
> Noted, will be done

Is this still an active PR ?

**Comment by HashZhang** [comment]
> > Noted, will be done
> 
> Is this still an active PR ?

Can merge this pull request and i will start another PR for .cmd modification

**Comment by HashZhang** [comment]
@RongtongJin @imaffe cmd file modified. Please review.

File: distribution/bin/runbroker.cmd [hunk 1]
```diff
@@ -28,6 +28,13 @@ set CLASSPATH=.;%BASE_DIR%conf;%CLASSPATH%
 rem ===========================================================================================
 rem  JVM Configuration
 rem ===========================================================================================
+
+for /f tokens^=2-5^ delims^=.-_^" %%j in ('java -fullversion 2^>^&1') do @set "JAVA_VERSION=%%j"
+rem 1代表1.8及以前的版本，其他代表9及以后的版本
```
Comment (vongosling) [inline distribution/bin/runbroker.cmd:33]: Pls remove the Chinese comment

File: distribution/bin/tools.cmd [hunk 2]
```diff
@@ -25,11 +25,23 @@ for %%d in (%BASE_DIR%) do set BASE_DIR=%%~dpd
 
 set CLASSPATH=.;%BASE_DIR%conf;%CLASSPATH%
 
+for /f tokens^=2-5^ delims^=.-_^" %%j in ('java -fullversion 2^>^&1') do @set "JAVA_VERSION=%%j"
+rem 1代表1.8及以前的版本，其他代表9及以后的版本
```
Comment (vongosling) [inline distribution/bin/tools.cmd:29]: So is it

File: distribution/bin/runbroker.cmd (hunk 1 above)
Comment (HashZhang) [reply distribution/bin/runbroker.cmd:33]: removed

File: distribution/bin/tools.cmd (hunk 2 above)
Comment (HashZhang) [reply distribution/bin/tools.cmd]: removed

**Comment by vongosling** [comment]
Please go ahead, I would like to see a rectangle chart that shows that this feature can pass on both the operating system and JDK platforms. In this way, we can merge into the trunk. 

Remember it we must test the jdk15 :-)

**Comment by HashZhang** [comment]
noted

**Pull request merged**